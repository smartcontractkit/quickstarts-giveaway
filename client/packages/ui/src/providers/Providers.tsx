import React from 'react'
import { Router } from 'react-router-dom'
import { createBrowserHistory } from 'history'
import { ChakraProvider } from '@chakra-ui/react'

import { theme } from '@ui/styles/theme'
import { Web3WalletProvider } from './'

const history = createBrowserHistory()

export const Providers = ({ children }) => (
  <Router history={history}>
    <ChakraProvider theme={theme}>
      <Web3WalletProvider>{children}</Web3WalletProvider>
    </ChakraProvider>
  </Router>
)
