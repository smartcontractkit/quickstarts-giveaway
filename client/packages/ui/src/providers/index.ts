export * from './Providers'
export * from './Web3WalletProvider'
