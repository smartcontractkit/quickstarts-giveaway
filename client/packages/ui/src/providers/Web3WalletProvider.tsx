import React from 'react'
import { WagmiConfig, createClient, configureChains } from 'wagmi'
import { InjectedConnector } from 'wagmi/connectors/injected'
import { jsonRpcProvider } from 'wagmi/providers/jsonRpc'
import {
  mainnet,
  localhost,
  hardhat,
  goerli,
  sepolia,
  polygonMumbai
} from 'wagmi/chains'

const DEFAULT_CHAIN_ID = polygonMumbai.id

export const CHAINS = [
  {
    ...mainnet,
    rpcUrls: {
      public: {
        http: ['https://cloudflare-eth.com']
      },
      default: {
        http: ['https://cloudflare-eth.com']
      }
    }
  },
  {
    ...localhost,
    rpcUrls: {
      public: {
        http: ['https://staking-metrics-monitor-alpha.staging.linkpool.io/dev']
      },
      default: {
        http: ['https://staking-metrics-monitor-alpha.staging.linkpool.io/dev']
      }
    }
  },
  {
    ...sepolia,
    rpcUrls: {
      public: {
        http: ['https://ethereum-sepolia-rpc.allthatnode.com']
      },
      default: {
        http: ['https://ethereum-sepolia-rpc.allthatnode.com']
      }
    }
  },
  {
    ...polygonMumbai,
    rpcUrls: {
      public: {
        http: ['https://endpoints.omniatech.io/v1/matic/mumbai/public']
      },
      default: {
        http: ['https://endpoints.omniatech.io/v1/matic/mumbai/public']
      }
    }
  },
  hardhat,
  goerli
]

const { chains, provider } = configureChains(
  CHAINS.sort((chain) => {
    return chain.id === DEFAULT_CHAIN_ID ? -1 : 1
  }),
  [
    jsonRpcProvider({
      priority: 1,
      rpc: (chain) => {
        return {
          http: chain.rpcUrls.default.http[0],
          webSocket: chain.rpcUrls.default.webSocket?.[0]
        }
      }
    }),
    jsonRpcProvider({
      rpc: (chain) => {
        if (chain.id !== 1) return null
        return { http: 'https://rpc.flashbots.net/' }
      }
    }),
    jsonRpcProvider({
      rpc: (chain) => {
        if (chain.id !== 1) return null
        return {
          http: 'https://api.mycryptoapi.com/eth'
        }
      }
    })
  ],
  { targetQuorum: 1 }
)

const connectors = () => {
  const list = [
    new InjectedConnector({
      chains
    })
  ]
  return list
}

export const client = createClient({
  autoConnect: true,
  connectors,
  provider
})

export const Web3WalletProvider = ({ children }) => (
  <WagmiConfig client={client}>{children}</WagmiConfig>
)
