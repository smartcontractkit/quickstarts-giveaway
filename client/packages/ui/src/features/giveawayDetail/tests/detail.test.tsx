import React from 'react'
import renderer from 'react-test-renderer'

import { Providers } from '@ui/providers'
import { GiveawayDetail } from '..'

const getComponent = () => {
  const component = (
    <Providers>
      <GiveawayDetail id={1} />
    </Providers>
  )

  return component
}

describe('GiveawayDetail', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(getComponent()).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
