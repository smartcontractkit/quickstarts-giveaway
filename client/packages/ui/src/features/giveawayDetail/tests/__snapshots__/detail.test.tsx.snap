// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiveawayDetail matches snapshot 1`] = `
[
  <span />,
  <span
    hidden={true}
    id="__chakra_env"
  />,
]
`;
